src/common/meta/src/key/table_name.rs

```diff
@@ -248,16 +248,23 @@ impl TableNameManager {
             .transpose()
     }
 
-    pub async fn tables(&self, catalog: &str, schema: &str) -> Result<Vec<String>> {
+    pub async fn tables(
+        &self,
+        catalog: &str,
+        schema: &str,
+    ) -> Result<Vec<(String, TableNameValue)>> {
         let key = TableNameKey::prefix_to_table(catalog, schema).into_bytes();
         let req = RangeRequest::new().with_prefix(key);
         let resp = self.kv_backend.range(req).await?;
-        let table_names = resp
-            .kvs
-            .into_iter()
-            .map(|kv| TableNameKey::strip_table_name(kv.key()))
-            .collect::<Result<Vec<_>>>()?;
-        Ok(table_names)
+
+        let mut res = Vec::with_capacity(resp.kvs.len());
+        for kv in resp.kvs {
+            res.push((
+                TableNameKey::strip_table_name(kv.key())?,
+                TableNameValue::try_from_raw_value(kv.value)?,
+            ))
+        }
```

Comment:
Can we use `into_iter` and `map`?

Reply:
`?` cannot be used in `map` and `collect<Result<>>` only works with one `Result` on the outer side. We could write a helper function but it seems trivial to do so.